lisa/sut_orchestrator/azure/common.py

```diff
@@ -243,13 +429,121 @@ def __hash__(self) -> int:
+        elif security_profile in (
+            "TrustedLaunchAndConfidentialVmSupported",
+            "ConfidentialVmSupported",
+        ):
+            capabilities.append(SecurityProfileType.SecureBoot)
+            capabilities.append(SecurityProfileType.CVM)
+            capabilities.append(SecurityProfileType.Stateless)
+        self.security_profile = security_profile = search_space.SetSpace(
+            True, capabilities
+        )
+
+    def _parse_version(self, platform: "AzurePlatform") -> "SharedImageGallerySchema":
+        compute_client = get_compute_client(platform)
+        if not self.resource_group_name:
+            # /subscriptions/xxxx/resourceGroups/xxxx/providers/Microsoft.Compute/
+            # galleries/xxxx
+            rg_pattern = re.compile(r"resourceGroups/(.*)/providers", re.M)
+            galleries = compute_client.galleries.list()
+            for gallery in galleries:
+                if gallery.name and gallery.name.lower() == self.image_gallery:
+                    assert gallery.id, "'gallery.id' must not be 'None'"
+                    self.resource_group_name = get_matched_str(gallery.id, rg_pattern)
+                    break
+        if not self.resource_group_name:
+            raise LisaException(f"did not find matched gallery {self.image_gallery}")
+
+        if self.image_version.lower() == "latest":
+            gallery_images = (
+                compute_client.gallery_image_versions.list_by_gallery_image(
+                    resource_group_name=self.resource_group_name,
+                    gallery_name=self.image_gallery,
+                    gallery_image_name=self.image_definition,
+                )
+            )
+            time: Optional[datetime] = None
+            for image in gallery_images:
+                assert image, "'image' must not be 'None'"
+                assert image.name, "'image.name' must not be 'None'"
+                gallery_image = compute_client.gallery_image_versions.get(
+                    resource_group_name=self.resource_group_name,
+                    gallery_name=self.image_gallery,
+                    gallery_image_name=self.image_definition,
+                    gallery_image_version_name=image.name,
+                    expand="ReplicationStatus",
+                )
+                if not time:
+                    time = gallery_image.publishing_profile.published_date
+                    assert image, "'image' must not be 'None'"
+                    assert image.name, "'image.name' must not be 'None'"
+                    self.image_version = image.name
+                elif gallery_image.publishing_profile.published_date > time:
+                    time = gallery_image.publishing_profile.published_date
+                    assert image, "'image' must not be 'None'"
+                    assert image.name, "'image.name' must not be 'None'"
+                    self.image_version = image.name
+        return copy.copy(self)
```

Comment:
Because the `self` is modified already, so it doesn't need to return a new object. An inplace update is enough.

Reply:
The problem I ran into with modifying self is that it doesn't call the setter method in AzureNodeSchema, so the object and the raw info become unsynced. It caused problems in other areas.

Reply:
It looks a copy doesn't solve the problem, which unsync between object the raw info. It should be solved at the caller level.